components/signup/signup_email/signup_email.tsx

```diff
@@ -23,7 +26,43 @@ import SiteNameAndDescription from 'components/common/site_name_and_description'
 
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 
-export default class SignupEmail extends React.PureComponent {
+import {Actions} from 'components/signup/signup_email/index';
+
+export type PasswordConfig = {
+    minimumLength: number;
+    requireLowercase: boolean;
+    requireNumber: boolean;
+    requireSymbol: boolean;
+    requireUppercase: boolean;
+}
+
+export type Props = {
+    location?: {search: string};
+    enableSignUpWithEmail: boolean;
+    siteName: string | '';
```

Comment:
Sorry, I meant that you could set siteName to `config.SiteName || ''` in mapStateToProps. We don't need to use the union type here since this just says it can be a string or the empty string which is redundant.
```suggestion
    siteName: string;
```